vmm/src/vm_config.rs

```diff
@@ -697,6 +699,72 @@ pub struct PayloadConfig {
     #[cfg(feature = "sev_snp")]
     #[serde(default)]
     pub host_data: Option<String>,
+    #[cfg(feature = "fw_cfg")]
+    pub fw_cfg_config: Option<FwCfgConfig>,
+}
+
+#[cfg(feature = "fw_cfg")]
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct FwCfgConfig {
+    #[serde(default)]
+    pub enabled: bool,
+    #[serde(default)]
+    pub e820: bool,
+    #[serde(default)]
+    pub kernel: bool,
+    #[serde(default)]
+    pub cmdline: bool,
+    #[serde(default)]
+    pub initramfs: bool,
+    #[serde(default)]
```

Comment:
I believe `serde` will default to `false`, which is not the behavior with CLI (which default to `true`).